src/Microsoft.Azure.SignalR.Management/NegotiationOptions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Connections;
+
+namespace Microsoft.Azure.SignalR.Management
+{
+    public class NegotiationOptions
+    {
+        /// <summary>
+        /// Gets or sets the HTTP context object that might provide information for routing and generating access token.
+        /// </summary>
+        public HttpContext HttpContext { get; set; }
+
+        /// <summary>
+        /// Gets or sets the user ID. If null, the identity name in <see cref="HttpContext.User" /> of the property <see cref="HttpContext"/> will be used.
+        /// </summary>
+        public string UserId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the claim list to be put into access token. If null, the claims in <see cref="HttpContext.User"/> of the property <see cref="HttpContext"/> will be used.
+        /// </summary>
+        public IList<Claim> Claims { get; set; }
+
+        /// <summary>
+        /// Gets or sets the lifetime of <see cref="NegotiationResponse.AccessToken"/>. Default value is one hour.
+        /// </summary>
+        public TimeSpan Lifetime { get; set; } = TimeSpan.FromHours(1);
+
+        /// <summary>
+        /// Gets or sets the flag indicates whether the client is a diagnostic client.
+        /// </summary>
+        public bool IsDiagnosticClient { get; set; } = false;
+
+        /// <summary>
+        /// Notifies when the negotiation operation should be cancelled. If null, the <see cref="HttpContext.RequestAborted"/> of <see cref="HttpContext"/> will be used.
+        /// </summary>
+        public CancellationToken CancellationToken { get; set; }
```

Comment:
this should be a parameter for the async method. Generally, every async method has a cancellation token to interpret the process